src/fsharp/IlxGen.fs

```diff
@@ -5649,8 +5671,14 @@ and ComputeMethodImplAttribs cenv (_v: Val) attrs =
     let hasAggressiveInliningImplFlag = (implflags &&& 0x0100) <> 0x0
     hasPreserveSigImplFlag, hasSynchronizedImplFlag, hasNoInliningImplFlag, hasAggressiveInliningImplFlag, attrs
 
-and GenMethodForBinding
-        cenv cgbuf eenv
+and GenMethodForBinding cenv (mgbuf: AssemblyBuilder) eenv info1 info2 =
+    if cenv.exprRecursionDepth > StackGuard.MaxUncheckedRecursionDepth then
```

Comment:
This is an interesting additional technique and we should pause to consider its ramifications

The main thing I see is that order of code generation might change, with some nested constructs being pushed to later.

Otherwise the IlxGen code is, I believe, quite robust to changing generation order.  But some testing will be in order to make sure this stays the case

Reply:
The order of the generation will certainly change. The thing to worry about here is `cenv` and `eenv` in this case. Right now, `cenv` is required when executing delayed gens, so I think that should be enough to cover any issues. `eenv` isn't required, which could be a problem. `eenv` may have context that isn't necessary or valid when generating a method in the middle of generating of another. I guess that's happening today too.

Reply:
Seems I was wrong on this. We do need to capture `eenv` as it contains context that is necessary at that point for generation.